src/main/java/ro/utcn/sd/agui/a1/controller/ConsoleController.java

```diff
@@ -0,0 +1,184 @@
+
+        System.out.println("Welcome to Stack Overflow by Andreea Gui!");
+
+        boolean done = false;
+
+        while(!done){
+
+            System.out.println("Enter a command: ");
+            String command = scanner.nextLine().trim();
+            try{
+                done = handleCommand(command);
+
+            } catch (UserNotFoundException e){
+                System.out.println("User was not found!");
+            } catch (QuestionNotFoundException e){
+                System.out.println("Question was not found!");
+            }catch (TagNotFoundException e){
+                System.out.println("Tag was not found.");
+            }
+
+        }
+
+    }
+
+    private boolean handleCommand(String command) {
+
+        switch (command){
+            case "login":
+                handleLogin();
+                return false;
+            case "logout":
+                handleLogout();
+                return false;
+            case "exit":
+                return true;
+        }
+
+        if (currentUser == null){
+            System.out.println("You need to be logged in to issue commands! ");
+            return false;
+        }
+
+        System.out.println("Your command is: " + command);
+        switch (command) {
+            case "list users":
+                handleListUsers();
+                return false;
+            case "list questions":
+                handleListQuestions();
+                return false;
+            case "register as user":
+                handleRegisterAsUser();
+                return false;
+            case "logout":
+                handleLogout();
+                return false;
+            case "add question":
+                handleAddQuestion();
+                return false;
+            case "filter questions by tag":
+                handleFilterQuestionsByTag();
+                return false;
+            case "filter questions by title":
+                handleFilterQuestionsByTitle();
+                return false;
+            case "exit":
+                return true;
+            default:
+                System.out.println("Unknown command. Please try again! ");
+                return false;
+        }
+    }
+
+    private void handleListUsers(){
+        userManagementService.listAllUsers().forEach(s->System.out.println(s.toString()));
+    }
+
+    private void handleListQuestions(){
+        questionManagementService.listAllQuestions().forEach(s->System.out.println(s.toString()));
+    }
+
+    private void handleRegisterAsUser(){
+        System.out.println("Username: ");
+        String username = scanner.nextLine();
+        System.out.println("Password: ");
+        String password = scanner.nextLine();
+        boolean registrationHappened = userManagementService.registerAsUser(username, password);
+        if(registrationHappened){
+            System.out.println("New user " + username + " was registered.");
+        }
+        else
```

Comment:
I strongly suggest that you get used to using the code formatter of your IDE. Usually developers use it as a reflex once they finish writing a method / block of code.  https://www.jetbrains.com/help/idea/reformat-and-rearrange-code.html